### PR TITLE
Search integration

### PR DESCRIPTION
**Pull request by charlie-wild**
## Link to JIRA ticket

https://trello.com/c/yezSWMVo/109-move-search-api-integration-to-sv-api

### *What changes have we introduced*

- Added Gateway, Use Case and Controller for integration with Housing Search API. 
- Added unit tests covering all new code additions

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: SingleViewApi/Startup.cs [hunk 1]
```diff
@@ -79,6 +79,20 @@ public void ConfigureServices(IServiceCollection services)
                 return new GetCustomerByIdUseCase(personGateway);
             });
 
+            services.AddTransient<IHousingSearchGateway, HousingSearchGateway>(s =>
+            {
+                var httpClient = s.GetService<IHttpClientFactory>().CreateClient();
+
+                return new HousingSearchGateway(httpClient,
+                    Environment.GetEnvironmentVariable("HOUSING_SEARCH_API_V1"));
```
Comment (elena-vi) [inline SingleViewApi/Startup.cs:87]: Let's add this env var to the sample

File: SingleViewApi/V1/Boundary/Response/HousingSearchApiResponseObject.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        public string DateOfBirth { get; set; }
+        public List<string> PersonTypes { get; set; }
+        public bool IsPersonCautionaryAlert { get; set; }
+        public bool IsTenureCautionaryAlert { get; set; }
+        public List<Tenure> Tenures { get; set; }
+    }
+
+    public class Results
+    {
+        public List<Person> Persons { get; set; }
```
Comment (elena-vi) [inline SingleViewApi/V1/Boundary/Response/HousingSearchApiResponseObject.cs:41]: Let's use `Person` from Hackney.Shared.Person

Comment (elena-vi) [inline SingleViewApi/V1/Boundary/Response/HousingSearchApiResponseObject.cs:41]: Let's use `Person` from Hackney.Shared.Person

File: SingleViewApi/V1/Boundary/Response/HousingSearchApiResponseObject.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Collections.Generic;
+
```
Comment (elena-vi) [inline SingleViewApi/V1/Boundary/Response/HousingSearchApiResponseObject.cs:3]:
```suggestion
using Hackney.Shared.Person;
```

Comment (elena-vi) [inline SingleViewApi/V1/Boundary/Response/HousingSearchApiResponseObject.cs:3]:
```suggestion
using Hackney.Shared.Person;
```

File: SingleViewApi/V1/Controllers/HousingSearchController.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            _getSearchResultsBySearchTextUseCase = getSearchResultsBySearchTextUseCase;
+        }
+
+        //TODO: add xml comments containing information that will be included in the auto generated swagger docs (https://github.com/LBHackney-IT/lbh-SingleViewApi/wiki/Controllers-and-Response-Objects)
+        /// <summary>
+        /// ...
+        /// </summary>
+        /// <response code="200">...</response>
+        /// <response code="400">Invalid Query Parameter.</response>
+        [ProducesResponseType(typeof(ResponseObjectList), StatusCodes.Status200OK)]
```
Comment (elena-vi) [inline SingleViewApi/V1/Controllers/HousingSearchController.cs:30]:
```suggestion
        [ProducesResponseType(typeof(SearchResponseObject, StatusCodes.Status200OK)]

```

Comment (elena-vi) [inline SingleViewApi/V1/Controllers/HousingSearchController.cs:30]:
```suggestion
        [ProducesResponseType(typeof(SearchResponseObject, StatusCodes.Status200OK)]

```

File: SingleViewApi/V1/UseCase/GetSearchResultsBySearchTextUseCase.cs [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+                        {
+                            Id = t.Id,
+                            CurrentAddress = t.IsActive,
+                            StartDate = t.StartDate,
+                            EndDate = t.EndDate,
+                            FullAddress = t.AssetFullAddress
+                        }))
+                    };
+
+                    Console.WriteLine("The person being added is {0}", person);
```
Comment (elena-vi) [inline SingleViewApi/V1/UseCase/GetSearchResultsBySearchTextUseCase.cs:64]: also this?

Comment (elena-vi) [inline SingleViewApi/V1/UseCase/GetSearchResultsBySearchTextUseCase.cs:64]: also this?

File: docker-compose.yml [hunk 6]
```diff
@@ -13,6 +13,7 @@ services:
     environment:
       - CONNECTION_STRING=Host=dev-database;Port=5432;Database=testdb;Username=postgres;Password=mypassword
       - PERSON_API_V1=${PERSON_API_V1}
+      - HOUSING_SEARCH_API_V1=${HOUSING_SEARCH_API_V1}
```
Comment (elena-vi) [inline docker-compose.yml:16]: this looks good, but we need to add this param in serverless.yml

**Review by elena-vi — COMMENTED** [review]
looks good, just need to reuse those predefined domain objects from `Hackney.Shared`

File: SingleViewApi/V1/UseCase/GetSearchResultsBySearchTextUseCase.cs [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+            {
+                housingSearchApiId.Error = "No results found";
+            }
+            else
+            {
+                var personResults = new List<SearchResult>();
+
+                foreach (var result in searchResults.Results.Persons)
+                {
+                    Console.WriteLine("Looping through search results - total results {0}", searchResults.Total);
```
[the remark this answers is not in the document]
Comment (charlie-wild) [reply SingleViewApi/V1/UseCase/GetSearchResultsBySearchTextUseCase.cs:42]: we do not!

**Review by elena-vi — APPROVED** [review]
Have you set the env vars in aws?

**Pull request merged**